From 80364a3c4b4bfe8353a9b03e06ec56995214d975 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kacper=20Michaj=C5=82ow?= <kasper93@gmail.com>
Date: Fri, 16 Oct 2026 17:27:40 +0200
Subject: [PATCH] avformat/dashdec: free the content of BaseURL elements
 starting with a slash
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

resolve_content_path() only freed the content of the elements it rewrote.

Signed-off-by: Kacper Michajłow <kasper93@gmail.com>
(cherry picked from commit 80364a3c4b4bfe8353a9b03e06ec56995214d975)
---
 libavformat/dashdec.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavformat/dashdec.c b/libavformat/dashdec.c
index 5e8efbcc470..cf88d86b0a3 100644
--- a/libavformat/dashdec.c
+++ b/libavformat/dashdec.c
@@ -846,6 +846,8 @@ static int resolve_content_path(AVFormatContext *s, const char *url, int *max_ur
             xmlNodeSetContent(baseurl_nodes[i], escaped);
             updated = 1;
             xmlFree(escaped);
+        } else {
+            xmlFree(text);
         }
     }
 
-- 
2.53.0

